keras_lr_finder/lr_finder.py

```diff
@@ -62,6 +62,36 @@ def find(self, x_train, y_train, start_lr, end_lr, batch_size=64, epochs=1):
         # Restore the original learning rate
         K.set_value(self.model.optimizer.lr, original_lr)
 
+    def find_generator(self, generator, start_lr, end_lr, epochs=1, num_batches=None):
+            if num_batches is None:
+                num_batches = len(generator)
+            self.lr_mult = (float(end_lr) / float(start_lr)) ** (float(1) /float(num_batches))
+
+            # Save weights into a file
+            self.model.save_weights('tmp.h5')
+
+            # Remember the original learning rate
+            original_lr = K.get_value(self.model.optimizer.lr)
+
+            # Set the initial learning rate
+            K.set_value(self.model.optimizer.lr, start_lr)
+
+            callback = LambdaCallback(on_batch_end=lambda batch,
+                                      logs: self.on_batch_end(batch, logs))
+
+            self.model.fit_generator(generator=generator,
+                                     epochs=epochs,
+                                     steps_per_epoch=num_batches,
+                                     use_multiprocessing=True,
```

Comment:
It might be good to make use_multiprocessing and workers configurable, as different use cases might require different settings.

Reply:
You are absolutely right. I moved them the multiprocessing part into optional kw arguments.